src/layout/Tooltip.js

```diff
@@ -0,0 +1,141 @@
+"use strict";
+(function (root, factory) {
+    if (typeof define === "function" && define.amd) {
+        define(["d3", "../common/HTMLWidget", "../layout/Popup", "../other/Table", "css!./Tooltip"], factory);
+    } else {
+        root.layout_Tooltip = factory(root.d3, root.common_HTMLWidget, root.layout_Popup, root.chart_Table);
+    }
+}(this, function (d3, HTMLWidget, Popup, Table) {
+    function Tooltip() {
+        HTMLWidget.call(this);
+        this._tag = "div";
+    }
+    Tooltip.prototype = Object.create(HTMLWidget.prototype);
+    Tooltip.prototype._class += " layout_Tooltip";
+    
+    Tooltip.prototype.publish("tooltipState", false, "boolean", "State of the popup, visible (true) or hidden (false)",null,{});
+    Tooltip.prototype.publish("offsetY", 10, "number", "Vertical offset from the cursor",null,{});
+    Tooltip.prototype.publish("offsetX", 10, "number", "Horizontal offset from the cursor",null,{});
+    Tooltip.prototype.publish("top", null, "number", "CSS top property",null,{});
+    Tooltip.prototype.publish("left", null, "number", "CSS left property",null,{});
+    Tooltip.prototype.publish("position", "fixed", "set", "Value of the 'position' property",["absolute", "relative", "fixed", "static", "initial", "inherit" ],{tags:['Private']});
+    Tooltip.prototype.publish("widget", null, "widget", "Widget",null,{tags:['Private']});
+
+    Tooltip.prototype.showTooltip = function(data, columns) {
+        this._table
+            .columns(columns)
+            .data([data])
+            .render()
+        ;
+        this.tooltipState(true).render();
+    };
+    
+    Tooltip.prototype.hideTooltip = function(data) {
+        this.tooltipState(false).render();
+    };
+    
+    Tooltip.prototype.registerWidget = function(widget) {
+        var context = this;
+        widget
+            .on("mouseover", function(d, i) {
+                var data = [],
+                    columns = [];
+                columns.push(widget._columns[0], i);
+                data.push(d[widget._columns[0]], d[i]);
+                context.showTooltip(data, columns);
+            })
+            .on("mouseout", function() {
+                context.hideTooltip();
+            })
+            .on("mousemove", function() {
+                var table = d3.select(context._table.node()).select("table"),
+                    height = table.node().offsetHeight,
+                    width = table.node().offsetWidth,
+                    rect;
+                
+                if (widget._parentRelativeDiv) { //This is because charts do not get assigned this when they are inside the MultiChart
+                    rect = widget._parentRelativeDiv.node().getBoundingClientRect();
+                } else {
+                    rect = widget._parentWidget._parentRelativeDiv.node().getBoundingClientRect();
```

Comment:
Going up 1 level will not always work as you can have SVG widgets inside SVG widgets etc.  I think you are looking for the rect of the widget in the same coordinate space as the tooltip? (so you prevent it from being clipped on the right/bottom)?